vividus-http-client/src/main/java/org/vividus/http/client/HttpClientConfig.java

```diff
@@ -75,7 +79,12 @@ public void setBaseUrl(String baseUrl)
 
     public boolean hasCredentials()
     {
-        return credentials != null && !credentials.isEmpty();
+        if (username != null && password != null)
+        {
+            return true;
+        }
+        isTrue(username == null && password == null, "The %s is missing", username == null ? "username" : "password");
```

Comment:
also when `preemptiveAuthenticationEnabled` is `true` and `username`/`password` is `null`, the state should treated as invalid